### PR TITLE
Disable default filetype mappings

### PR DESCRIPTION
**Pull request by russtoku**
On 06/04/23, `@nmkip` asked on [Discord:](https://discord.com/channels/732957595249410108/732964914586714163/1114941033931866152):
> Is there a way to disable all the default mappings in one line or do I have to go one by one?

This PR adds an option to disable all of the default mappings for the clients listed in `g:conjure#filetypes`.

I hope I didn't misunderstand what was wanted.

### CODE REVIEW

File: lua/conjure/client/guile/socket.lua [hunk 1]
```diff
@@ -157,7 +157,7 @@ local function parse_guile_result(s)
     if s:find("scheme@%([%w%-%s]+%) %[%d+%]>") then
       return {["done?"] = true, ["error?"] = true, result = nil}
     else
-      return {result = s, ["done?"] = false, ["error?"] = false}
+      return {result = s, ["error?"] = false, ["done?"] = false}
```
Comment (Olical) [inline lua/conjure/client/guile/socket.lua:160]: It's annoying that this order changes randomly :( would love to fix it some day. I tend to do `make` then I discard changes like this caused by fennel compiler instability, personally. It's not a big deal, but sometimes the config compiler output and this line specifically can change orders which I just ignore if I know I haven't changed them.

File: fnl/conjure/mapping.fnl [hunk 2]
```diff
@@ -206,9 +206,10 @@
 (defn init [filetypes]
   (nvim.ex.augroup :conjure_init_filetypes)
   (nvim.ex.autocmd_)
-  (nvim.ex.autocmd
-    :FileType (str.join "," filetypes)
-    (bridge.viml->lua :conjure.mapping :on-filetype {}))
+  (if (= true (config.get-in [:mapping :enable_ft_mappings]))
+    (nvim.ex.autocmd
+      :FileType (str.join "," filetypes)
+      (bridge.viml->lua :conjure.mapping :on-filetype {})))
```
[the remark this answers is not in the document]
Comment (russtoku) [reply fnl/conjure/mapping.fnl:212]: Yes, thanks for the tip about using `when` instead `if`.

File: lua/conjure/client/guile/socket.lua (hunk 1 above)
Comment (russtoku) [reply lua/conjure/client/guile/socket.lua:160]: Agreed, that's quite annoying. Seems to happen quite frequently but I didn't take a closer look at it.

File: fnl/conjure/mapping.fnl (hunk 2 above)
[the remark this answers is not in the document]
Comment (russtoku) [reply fnl/conjure/mapping.fnl:212]: Changed with commit [d0a9ac0](https://github.com/Olical/conjure/pull/502/commits/d0a9ac03a28980e2b2cd6f471ec727c0f873c7d1).

**Comment by russtoku** [comment]
BTW, I forgot to mention that I arbitrarily chose `conjure#mapping#enable_ft_mappings` and am not terribly wedded to it. Please choose something that better aligns with your liking.

**Comment by Olical** [comment]
Thanks a lot! I like the name, it's as good as any and I'd only pick something similar. Just a colour of a bike shed really 😄 much appreciated!

**Pull request merged**